src/output.c

```diff
@@ -1032,6 +1025,109 @@ void OutputRegisterRootLoggers(void)
     OutputStreamingLoggerRegister();
 }
 
+static int JsonGenericLogger(ThreadVars *tv, void *thread_data, const Packet *p, Flow *f,
+        void *state, void *tx, uint64_t tx_id)
+{
+    OutputJsonThreadCtx *thread = thread_data;
+    AppLayerLogger *al = GetAppProtoLogger(f->alproto);
+    if (al == NULL) {
+        return TM_ECODE_FAILED;
+    }
+
+    JsonBuilder *js = CreateEveHeader(p, LOG_DIR_PACKET, al->name, NULL, thread->ctx);
+    if (unlikely(js == NULL)) {
+        return TM_ECODE_FAILED;
+    }
+
+    jb_open_object(js, al->name);
+    if (!al->log(tx, js)) {
+        goto error;
```

Comment:
missing close on error?

Reply:
Likewise here, just free'd on error, not used anymore.